gremlin-go/driver/graphBinary.go

```diff
@@ -199,6 +201,90 @@ func mapReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return valMap, nil
 }
 
+func getSignedBytesFromBigInt(n *big.Int) []byte {
+	var one = big.NewInt(1)
+	switch n.Sign() {
+	case 0:
+		return []byte{}
+	case 1:
+		b := n.Bytes()
+		if b[0]&0x80 > 0 {
+			b = append([]byte{0}, b...)
+		}
+		return b
+	case -1:
+		length := uint(n.BitLen()/8+1) * 8
+		b := new(big.Int).Add(n, new(big.Int).Lsh(one, length)).Bytes()
+		if len(b) >= 2 && b[0] == 0xff && b[1]&0x80 != 0 {
+			b = b[1:]
+		}
+		return b
+	}
```

Comment:
This code is a little esoteric, could you add a comment about what you are doing? Ex: What does b[0]&0x80 give you in terms of information?

Also I suggest adjusting the code into a form where panic is not needed, example below:
```suggestion
	if (n.Sign() == 1) {
		b := n.Bytes()
		if b[0]&0x80 > 0 {
			b = append([]byte{0}, b...)
		}
		return b
	} else if (n.Sign() == -1) {
		length := uint(n.BitLen()/8+1) * 8
		b := new(big.Int).Add(n, new(big.Int).Lsh(one, length)).Bytes()
		if len(b) >= 2 && b[0] == 0xff && b[1]&0x80 != 0 {
			b = b[1:]
		}
		return b
	}
	return []byte{}
```

Reply:
Are we trying to avoid panics? Wouldn't debugging get harder if we're returning an empty byte array?

Reply:
Generally avoid panics, and in this case we have the following definition
![image](https://user-images.githubusercontent.com/58273576/153091267-1e7e6d5c-8f94-4a6a-86ef-985b51cebed7.png)
which tells us that it should never be not 1, -1, or 0. So let's assume it's 0 if it is not 1 or -1

Reply:
Oops that was for float, anyway it is the same for int

Reply:
My understanding is that `go` panics are used almost exclusively in case of catastrophic failure such that you need to kill the program. I think what we need is more of logging and an error return for traceability.